### PR TITLE
[Documentation] register_processed_data.pl perldocification

### PR DESCRIPTION
**Pull request by cmadjar**
Using perldoc/perlpod to document `register_processed_data.pl` so that users can type in the terminal
`perldoc register_processed_data.pl` and get the documentation.

Using the `pod2markdown` tool from perl `Pod::Markdown` module, the `register_processed_data.pl` file has been created so that we have a web displayed version of the documentation.
`pod2markdown register_processed_data.pl > register_processed_data.md`

Dependencies added: perldoc, Pod::Markdown, Pod::Usage

### CODE REVIEW

**Comment by MounaSafiHarab** [comment]
@cmadjar 
can you please update the top description of your pull request.

you have:
`pod2markdown register_processed_data.pl > register_processed_data.pl`

it should be
`pod2markdown register_processed_data.pl > register_processed_data.md`

This applies to many pull requests :)

**Comment by MounaSafiHarab** [comment]
hey @nicolasbrossard 

I took the liberty to assign some .pl documentation PRs to you (I split them between me and you, as we agreed on last time during our meeting with @cmadjar);
those eventually need to all be merged to the loris-mri-docs branch so that when 19.1-dev is created, we can proceed to merging the loris-mri-docs branch into minor ASAP after...

File: uploadNeuroDB/register_processed_data.pl [hunk 1]
```diff
@@ -276,9 +333,19 @@
...
 ###################################
+
 =pod
-This function returns the sessionID based on sourceFileID.
+
+=head3 getSessionID($sourceFileID, $dbh)
+
+This function returns the sessionID based on the provided sourceFileID.
+
+INPUT: source FileID, database handle
```
Comment (nicolasbrossard) [inline uploadNeuroDB/register_processed_data.pl]: I guess if we want to be consistent with other PRs, this should be:
```perl
INPUTS:
    - source FileID.
    - database handle.
```

Or maybe this is also valid. What do you think?

This comment applies to all `INPUT` sections.

File: uploadNeuroDB/register_processed_data.pl [hunk 2]
```diff
@@ -276,9 +333,19 @@
...
+
+RETURNS: session ID
```
Comment (nicolasbrossard) [inline uploadNeuroDB/register_processed_data.pl]: Consistency: `RETURNS` or `RETURN`? Applies to the whole script.

File: uploadNeuroDB/register_processed_data.pl [hunk 3]
```diff
@@ -311,8 +378,17 @@ sub getSessionID    {
 
 
 =pod
-This function gets ScannerID from files using sourceFileID
+
+=head3 getScannerID($sourceFileID, $dbh)
+
+This function gets ScannerID from the C<files> table using sourceFileID
```
Comment (nicolasbrossard) [inline uploadNeuroDB/register_processed_data.pl]: This function gets _the_ ScannerID

File: uploadNeuroDB/register_processed_data.pl [hunk 4]
```diff
@@ -354,9 +441,19 @@ sub getAcqProtID    {
     return  ($acqProtID);
 }
 
+
 =pod
-This function parses the mincheader and look for specific field's value.
+
+=head3 fetchMincHeader($file, $field)
+
+This function parses the MINC header and look for specific field's value.
```
Comment (nicolasbrossard) [inline uploadNeuroDB/register_processed_data.pl]: ....header and _looks_ for specific...

File: uploadNeuroDB/register_processed_data.pl [hunk 5]
```diff
@@ -1,5 +1,60 @@
 #!/usr/bin/perl -w
 
+=pod
+
+=head1 NAME
+
+register_processed_data.pl -- Inserts processed data and link it to the source
```
Comment (nicolasbrossard) [inline uploadNeuroDB/register_processed_data.pl]: ...data and _links_ it to....

File: uploadNeuroDB/register_processed_data.pl (hunk 4 above)
Comment (nicolasbrossard) [inline uploadNeuroDB/register_processed_data.pl]: ...for specific _field values_

File: uploadNeuroDB/register_processed_data.pl [hunk 6]
```diff
@@ -465,13 +586,22 @@ sub which_directory {
...
-Function that will insert into the files_intermediary table of the database, intermediary outputs that were used to obtain the processed file.
-- Input:  - fileID : fileID of the registered processed file
-          - inputFileIDs : array containing the list of input files that were used to obtain the processed file
-          - tool : tool that was used to obtain the processed file
-- Output: - return undef if insertion did not succeed
-          - return 1 if insertion into the intermediary table succeeded. 
+
+=head3 insert_intermedFiles($fileID, $inputFileIDs, $tool)
+
+Function that will insert into the C<files_intermediary> table of the database,
```
Comment (nicolasbrossard) [inline uploadNeuroDB/register_processed_data.pl]: Grammar: Function that will insert the intermediary outputs that were used to obtain the processed file into the C<files_intermediary> table of the database.

File: uploadNeuroDB/register_processed_data.pl (hunk 5 above)
Comment (cmadjar) [reply uploadNeuroDB/register_processed_data.pl]: done

File: uploadNeuroDB/register_processed_data.pl (hunk 1 above)
Comment (cmadjar) [reply uploadNeuroDB/register_processed_data.pl]: you're absolutely right! Fixing them.

File: uploadNeuroDB/register_processed_data.pl (hunk 3 above)
Comment (cmadjar) [reply uploadNeuroDB/register_processed_data.pl]: done

File: uploadNeuroDB/register_processed_data.pl (hunk 4 above)
Comment (cmadjar) [reply uploadNeuroDB/register_processed_data.pl]: done

Comment (cmadjar) [reply uploadNeuroDB/register_processed_data.pl]: done

**Comment by cmadjar** [comment]
@nicolasbrossard Ready for another round of review :)

File: uploadNeuroDB/register_processed_data.pl [hunk 7]
```diff
@@ -354,9 +447,21 @@ sub getAcqProtID    {
...
 }
 
+
 =pod
-This function parses the mincheader and look for specific field's value.
+
+=head3 fetchMincHeader($file, $field)
+
+This function parses the MINC header and looks for specific field's value.
+
```
Comment (nicolasbrossard) [inline uploadNeuroDB/register_processed_data.pl]: ..and looks for a specific field value

Comment (cmadjar) [reply uploadNeuroDB/register_processed_data.pl]: @nicolasbrossard Thanks! Sorry I missed it. It has been changed.

**Comment by MounaSafiHarab** [comment]
@nicolasbrossard 

your feedback were incorporated by @cmadjar 
can you please take a final look, and merge if all is good?
removing the Needs work tag.

**Pull request merged**